src/nnet3/nnet-simple-component.cc

```diff
@@ -4780,8 +4751,19 @@ std::string CompositeComponent::Info() const {
 
 // virtual
 void CompositeComponent::Scale(BaseFloat scale) {
-  for (size_t i = 0; i < components_.size(); i++)
-    components_[i]->Scale(scale);
+  if (scale == 0.0) {
```

Comment:
I don't think you have to modify the code of this function, just leave it as it is.